q2cli/util.py

```diff
@@ -73,6 +77,43 @@ def exit_with_error(e, header='An error has been encountered:',
     click.get_current_context().exit(status)
 
 
+def output_in_cache(fp):
+    """Determines if an output path follows the format
+    /path_to_extant_cache:key
+    """
+    from pathlib import Path
+    from qiime2.core.cache import Cache
+
+    # Tells us right away this isn't in a cache
+    if ':' not in fp:
+        return False
+
+    split_path = fp.split(':')
+
+    # Account for potential for : in the path
+    cache_path = ':'.join(elem for elem in split_path[:-1])
+    key = split_path[-1]
```

Comment:
An easier way to do this and the split is:
```python
cache_path, key = fp.rsplit(':', 1)
```
Which will split from the right, at most once. The unpacking should also always succeed as we have proved there is at least 1 colon in the path.